ReactiveXComponent/RabbitMq/RabbitMqSubscriber.cs

```diff
@@ -19,112 +19,128 @@ public class RabbitMqSubscriber : IXCSubscriber
         private readonly IXCConfiguration _xcConfiguration;
         private readonly string _privateCommunicationIdentifier;
 
-        private readonly Dictionary<SubscriberKey, RabbitMqSubscriberInfos> _subscribers;
-        private readonly Dictionary<SubscriberKey, List<Action<MessageEventArgs>>> _listenerToUpdateBySubscriberKeyRepo;
+        private readonly ConcurrentDictionary<SubscriberKey, RabbitMqSubscriberInfos> _subscribers;
+        private readonly ConcurrentDictionary<SubscriberKey, ThreadSafeList<Action<MessageEventArgs>>> _listenerToUpdateBySubscriberKeyRepo;
```

Comment:
I think that you should simply manage thread-safety in the methods where you access the listeners dictionary. there is no need for the ThreadSafeList.